test/test_verify.py

```diff
@@ -198,8 +198,43 @@ def test_verify_leaf_certs_no_eku(
         ):
             verifier._verify_leaf_certs(tsp_response=ts_response)
 
-    def test_verify_leaf_cert_mismatch(
+    def test_verify_leaf_certs_non_critical_eku(
+        self, verifier: Verifier, ts_response: TimeStampResponse, monkeypatch: MonkeyPatch
+    ) -> None:
+        monkeypatch.setattr(cryptography.x509.Extension, "critical", False)
+        with pytest.raises(VerificationError, match="The EKU extension is not critical"):
+            verifier._verify_leaf_certs(tsp_response=ts_response)
+
+    def test_verify_leaf_certs_eku_no_values(
+        self, verifier: Verifier, ts_response: TimeStampResponse, monkeypatch: MonkeyPatch
+    ) -> None:
+        monkeypatch.setattr(
+            cryptography.x509.Extension,
+            "value",
+            [],
+        )
+        with pytest.raises(
+            VerificationError, match="The EKU extension does not have any KeyPurposeID"
+        ):
+            verifier._verify_leaf_certs(tsp_response=ts_response)
+
+    def test_verify_leaf_certs_non_time_stamping_eku(
         self, verifier: Verifier, ts_response: TimeStampResponse, monkeypatch: MonkeyPatch
+    ) -> None:
+        monkeypatch.setattr(
+            cryptography.x509.Extension,
+            "value",
+            # this is a random OID, not id-kp-timeStamping
+            [cryptography.x509.ObjectIdentifier("1.9.9.9.9.9.9.9.9")],
```

Comment:
Rather than a maybe-valid OID, this one is guaranteed to be nonexistent:

```suggestion
            [cryptography.x509.ObjectIdentifier("1.3.6.1.4.1.55738.666.4")],
```

(I control that OID's arc so I can say define its semantics)

Reply:
Since the refactor, we no longer need this test (but I'll keep in mind this OIDs :))